src/modules/flow/ipm/ipm.c

```diff
@@ -0,0 +1,599 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "sol-flow/ipm.h"
+#include "sol-flow-internal.h"
+#include "sol-ipm.h"
+
+#include <sol-util.h>
+#include <errno.h>
+
+/* TODO would be lovely to have this available globally... create a new
+* blob from some variable - but duplicating it, not just wrapping it */
+#define CREATE_BLOB(_val) \
```

Comment:
add these helpers to sol-blob.h, makes sense. But make a static inline function that does the work and a macro that does the sizeof() and calls it.